ethereum/README.md

```diff
@@ -2,61 +2,101 @@
 
 This directory contains smart contracts utilized by the Polkadot-Ethereum Bridge.
 
+- Bridge.sol: application registry and routing of messages from Substrate
+- Application.sol: an abstract contract that must be implemented by any Bridge application
+    - ETHApp.sol: application for cross-chain ETH transfers between Ethereum and Substrate
+    - ERC20App.sol: application for cross-chain ERC20 transfers between Ethereum and Substrate
+- Decoder.sol: a library for decoding SCALE encoded data
+- Verifier.sol: verifies tx origin and signatures
+- Scale.sol: implements decoding of SCALE encoded compact uints; not currently used, will support future work on generalized data relay
+
 ## Set up
 
-Create a `.env` file using the following template:
+After starting the blockchain and deploying the contracts, the Bridge's Ethereum component is ready for usage. To generate configuration information required for configuring the Parachain and Relayer components, run the set up scripts described in scripts section below.
 
 ```bash
-cp .env.example .env
```

Comment:
I'm pretty sure I needed the toplevel `.env` file in order to run `truffle migrate`. That's what I encountered previously.

Reply:
Ah I believe this is true iff deploying to the ropsten testnet as it requires `MNEMONIC` and `INFURA_PROJECT_ID`

Reply:
Resolved in https://github.com/Snowfork/polkadot-ethereum/pull/107/commits/5d02b9d9f0e7f00a7a8082af1a854a096165dee6 - `.env` now lives at the top level and the scripts pull the config file from their parent directory.